### PR TITLE
rustdesk: 1.3.0 -> 1.3.1

### LINKED ISSUE #349704: Update request: rustdesk 1.3.0 → 1.3.1
- Package name: `rustdesk`
- Latest released version: `1.3.1`
- Current version on the unstable channel: `1.3.0`
- Current version on the stable/release channel: `1.2.3`
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by khaneliman**
Resolves https://github.com/NixOS/nixpkgs/issues/349704

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by khaneliman** [comment]
nix-update seems to have downgraded a bunch of versions in the cargo lock. Might be nice to have a proper update script in this derivation.

**Comment by khaneliman** [comment]
@GrahamcOfBorg build

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349891`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

---
### `aarch64-linux`

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
```bash
 warning: `hbb_common` (lib) generated 3 warnings (1 duplicate)
┃        >    Compiling rustdesk v1.3.1 (/private/tmp/nix-build-rustdesk-1.3.1.drv-0/source)
┃        > error: failed to run custom build command for `magnum-opus v0.4.0 (https://github.com/rustdesk-org/magnum-opus#5cd2bf98)`
┃        >
┃        > Caused by:
┃        >   process didn't exit successfully: `/private/tmp/nix-build-rustdesk-1.3.1.drv-0/source/target/release/build/magnum-opus-baa076edece87af7/build-script-build` (exit status: 101)
┃        >   --- stderr
┃        >   thread 'main' panicked at /private/tmp/nix-build-rustdesk-1.3.1.drv-0/cargo-vendor-dir/magnum-opus-0.4.0/build.rs:64:9:
┃        >   Couldn't find VCPKG_ROOT, also can't fallback to homebrew because it's only for macos aarch64.
┃        >   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349891`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349891`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

---
### `aarch64-linux`

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
```bash
  cargo:warning=src/platform/macos.mm:1:9: fatal error: 'AVFoundation/AVFoundation.h' file not found
┃        >   cargo:warning=#import <AVFoundation/AVFoundation.h>
┃        >   cargo:warning=        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
┃        >   cargo:warning=1 error generated.
┃        >
┃        >   --- stderr
┃        >
┃        >
┃        >   error occurred: Command env -u IPHONEOS_DEPLOYMENT_TARGET "/nix/store/22a6p8311lg3z4ksrjbj018i7z3vgkva-clang-wrapper-16.0.6/bin/cc" "-O3" "-ffunction-sections" "-fdata-sec…
┃        >

**Comment by Leixb** [comment]
@khaneliman I don't think we have libsciter packaged for darwin. Correct me if I am wrong, but even if you get it to build it will probably won't run.

**Comment by khaneliman** [comment]
> @khaneliman I don't think we have libsciter packaged for darwin. Correct me if I am wrong, but even if you get it to build it will probably not run.

Oh, that sounds familiar... I believe I started working on packaging libsciter a long time ago, but got sidetracked.

**Comment by Leixb** [comment]
Result of `nixpkgs-review pr 349891` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

**Review by Leixb — APPROVED** [review]

File: pkgs/by-name/ru/rustdesk/package.nix [hunk 1]
```diff
@@ -37,53 +37,55 @@
...
     };
   };
 
+  env.VCPKG_ROOT = "/homeless-shelter"; # idk man, makes the build go since https://github.com/21pages/hwcodec/commit/1873c34e3da070a462540f61c0b782b7ab15dc84
+
   postPatch = ''
     # Overwrite cargo.lock because the one in the upstream repo has duplicates entries.
     # It should probably be removed in the next rustdesk update (if they fix their cargoLock)
-    ln -fs ${./Cargo.lock} Cargo.lock
+    cp ${./Cargo.lock} Cargo.lock
```
Comment (drupol) [inline pkgs/by-name/ru/rustdesk/package.nix:88]: Why this change?

File: pkgs/by-name/ru/rustdesk/package.nix [hunk 2]
```diff
@@ -37,53 +37,55 @@
...
-      "tray-icon-0.15.1" = "sha256-DvTVAsE3dB+o89SFk3RPRt5JzaWjPl4nlvC77zsVN/s=";
+      "tray-icon-0.14.3" = "sha256-dSX7LucZaLplRrh6zLwmFzyZN4ZtwIXzAEdZzlu3gQg=";
       "wallpaper-3.2.0" = "sha256-p9NRmusdA0wvF6onp1UTL0/4t7XnEAc19sqyGDnfg/Q=";
       "webm-1.1.0" = "sha256-p4BMej7yvb8c/dJynRWZmwo2hxAAY96Qx6Qx2DbT8hE=";
       "x11-2.19.0" = "sha256-GDCeKzUtvaLeBDmPQdyr499EjEfT6y4diBMzZVEptzc=";
       "x11-clipboard-0.8.1" = "sha256-PtqmSD2MwkbLVWbfTSXZW3WEvEnUlo04qieUTjN2whE=";
     };
   };
 
+  env.VCPKG_ROOT = "/homeless-shelter"; # idk man, makes the build go since https://github.com/21pages/hwcodec/commit/1873c34e3da070a462540f61c0b782b7ab15dc84
```
Comment (drupol) [inline pkgs/by-name/ru/rustdesk/package.nix:83]: Please rewrite the comment.

File: pkgs/by-name/ru/rustdesk/package.nix (hunk 1 above)
Comment (khaneliman) [reply pkgs/by-name/ru/rustdesk/package.nix:88]: https://github.com/NixOS/nixpkgs/pull/349891#issuecomment-2424606562

File: pkgs/by-name/ru/rustdesk/package.nix (hunk 2 above)
Comment (khaneliman) [reply pkgs/by-name/ru/rustdesk/package.nix:83]: Sorry, forgot that was leftover from copying from the flutter derivation.

File: pkgs/by-name/ru/rustdesk/package.nix (hunk 1 above)
Comment (khaneliman) [reply pkgs/by-name/ru/rustdesk/package.nix:86]: Also matches what they had to do in rustdesk-flutter.

File: pkgs/by-name/ru/rustdesk/package.nix [hunk 3]
```diff
@@ -178,6 +178,7 @@ rustPlatform.buildRustPackage rec {
   env = {
     SODIUM_USE_PKG_CONFIG = true;
     ZSTD_SYS_USE_PKG_CONFIG = true;
+    VCPKG_ROOT = "/homeless-shelter"; # Build error of "Couldn't find VCPKG_ROOT" on darwin
```
Comment (khaneliman) [inline pkgs/by-name/ru/rustdesk/package.nix:181]: Could probably even pull this out until the darwin support is added.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349891`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rustdesk</li>
  </ul>
</details>

---
### `aarch64-linux`

---
### `x86_64-darwin`

---
### `aarch64-darwin`
